release/nightly_tests/dataset/operator_fusion_benchmark.py

```diff
@@ -0,0 +1,182 @@
+import argparse
+import json
+import time
+from typing import Dict, List, Any
+
+import pandas as pd
+import pyarrow as pa
+import numpy as np
+
+import ray
+from ray.data.block import BlockMetadata
+from ray.data.context import DatasetContext, DEFAULT_TARGET_MAX_BLOCK_SIZE
+from ray.data.datasource import Datasource, ReadTask, Reader
+
+
+class BlockDatasource(Datasource):
```

Comment:
Any reason we can't use the RangeDatasource?

Reply:
Primarily to control:
1. The number of blocks per read task.
2. The number of columns per block.

Since both of these are important dimensions of the benchmark.